src/device/device.hpp

```diff
@@ -104,7 +109,18 @@ namespace hw {
         virtual bool release() = 0;
 
         virtual bool connect(void) = 0;
-        virtual bool disconnect() = 0;
+        virtual bool disconnect(void) = 0;
+
+        virtual bool  set_mode(unsigned int mode) = 0;
```

Comment:
Should be device_mode now

Reply:
you mean  `bool  set_device_mode(unsigned int mode)` ?

Reply:
I mean set_mode(device_mode mode)

Reply:
ok